apps/mux/build-actions.js

```diff
@@ -0,0 +1,82 @@
+const esbuild = require('esbuild');
+const { resolve } = require('path');
+const yargs = require('yargs/yargs');
+const { hideBin } = require('yargs/helpers');
+
+const manifest = require('./contentful-app-manifest.json');
+
+const env = process.env.NODE_ENV;
+const path = `./app-actions/.env${env === 'development' ? `.${env}` : ''}`;
+require('dotenv').config({ path });
+
+const argv = yargs(hideBin(process.argv)).argv;
+
+const validateActions = () => {
+  const requiredProperties = ['id', 'path', 'entryFile'];
+  const uniqueValues = new Set();
+
+  manifest.actions.forEach((action) => {
+    requiredProperties.forEach((property) => {
+      if (!action.hasOwnProperty(property)) {
+        throw new Error(`Action with name: '${action.name}' is missing the '${property}' property`);
+      }
+    });
+
+    const { id, path, entryFile } = action;
+
+    if (uniqueValues.has(id)) {
+      throw new Error(`Duplicate action id: '${id}'`);
+    }
+    if (uniqueValues.has(path)) {
+      throw new Error(`Duplicate action path: '${path}'`);
+    }
+    if (uniqueValues.has(entryFile)) {
+      throw new Error(`Duplicate entryFile path: '${entryFile}'`);
+    }
+
+    uniqueValues.add(entryFile);
+    uniqueValues.add(path);
+    uniqueValues.add(id);
+  });
+};
+
+const getEntryPoints = () => {
+  return manifest.actions.reduce((result, action) => {
+    const fileName = action.path.split('.')[0];
+
+    result[fileName] = resolve(__dirname, action.entryFile);
+
+    return result;
+  }, {});
+};
+
+const main = async (watch = false) => {
+  try {
+    console.log('Building app actions');
```

Comment:
seeing a few` console.logs` throughout the file, but assume these will be removed when the dust settles!

Reply:
this is another one where the logs are intentional since this is really a build (CLI) tool. so the purpose of these is to get output on the screen when it's run. 😄